File: 프로그래머스/2/42578. 의상/의상.java
import java.util.*;
class Solution {
    public int solution(String[][] clothes) {
        int answer = 1;
        
        Map<String, Integer> map = new HashMap<>();
        for(String[] cloth : clothes){
            String type = cloth[1];
            map.put(type, map.getOrDefault(type, 0) + 1);
        }
        
        for (int values: map.values()){
            answer *= (values+1);
        }
        
        return answer-1;
    }
}

File: 프로그래머스/2/42578. 의상/의상.py
from collections import defaultdict
def solution(clothes):
    answer = 1
    
    c = defaultdict(int)
    for clothe_name, clothe_type in clothes:
        c[clothe_type] +=1
    
    for i in c.keys():
        answer = answer * (c[i]+1)
    return answer-1